### PR TITLE
feat(api-client): Add federated pre-key fetching

### PR DESCRIPTION
**Pull request by bennycode**

### CODE REVIEW

File: packages/api-client/src/user/UserAPI.ts [hunk 1]
```diff
@@ -40,6 +40,11 @@ import {
 import {RichInfo} from './RichInfo';
 import {RequestCancellationError} from './UserError';
 
+export type QualifiedUser = {
+  domain: string;
+  id: string;
+};
+
```
Comment (ffflorian) [inline packages/api-client/src/user/UserAPI.ts:47]: Let's move this interface to `User.ts`, WDYT?

File: packages/api-client/src/user/UserAPI.ts [hunk 2]
```diff
@@ -268,17 +273,22 @@ export class UserAPI {
...
-  /**
-   * Get a prekey for each client of a user.
-   * @param userId
-   */
-  public async getUserPreKeys(userId: string): Promise<PreKeyBundle> {
+  public async getUserPreKeys(user: QualifiedUser): Promise<QualifiedPreKeyBundle>;
+
+  public async getUserPreKeys(user: string): Promise<PreKeyBundle>;
+
+  public async getUserPreKeys(user: QualifiedUser | string): Promise<PreKeyBundle | QualifiedPreKeyBundle> {
```
Comment (ffflorian) [inline packages/api-client/src/user/UserAPI.ts:280]:
```suggestion
  public async getUserPreKeys(user: QualifiedUser): Promise<QualifiedPreKeyBundle>;
  public async getUserPreKeys(user: string): Promise<PreKeyBundle>;
  public async getUserPreKeys(user: QualifiedUser | string): Promise<PreKeyBundle | QualifiedPreKeyBundle> {
```

**Review by ffflorian — CHANGES_REQUESTED** [review]

File: packages/api-client/src/auth/PreKeyBundle.ts [hunk 3]
```diff
@@ -18,10 +18,19 @@
...
   clients: ClientPreKey[];
+};
+
+export type PreKeyBundle = PreKeyBundleBase & {
   /** The user ID */
   user: string;
-}
+};
+
+export type QualifiedPreKeyBundle = PreKeyBundleBase & {
```
Comment (ffflorian) [inline packages/api-client/src/auth/PreKeyBundle.ts:33]: What's the benefit of this style over
```ts
export interface QualifiedPreKeyBundle extends PreKeyBundleBase {
```
?

File: packages/api-client/src/user/UserAPI.ts (hunk 1 above)
Comment (bennycode) [reply packages/api-client/src/user/UserAPI.ts:47]: Ok

File: packages/api-client/src/auth/PreKeyBundle.ts (hunk 3 above)
Comment (bennycode) [reply packages/api-client/src/auth/PreKeyBundle.ts:33]: No benefit. I will stick with interfaces.

**Review by ffflorian — APPROVED** [review]

**Pull request merged**